cli/bptest/run_test.go

```diff
@@ -0,0 +1,103 @@
+package bptest
+
+import (
+	"fmt"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsValidTestName(t *testing.T) {
+	tests := []struct {
+		name       string
+		intTestDir string
+		testName   string
+		errMsg     string
+	}{
+		{
+			name:     "valid explicit",
+			testName: "TestBar",
+		},
+		{
+			name:     "valid discovered",
+			testName: "TestAll/examples/baz",
+		},
+		{
+			name:     "valid all regex",
+			testName: "Test.*",
+		},
+		{
+			name:     "all",
+			testName: "all",
+		},
+		{
+			name:     "invalid",
+			testName: "TestBaz",
+			errMsg:   "unable to find TestBaz- one of [\"TestAll/examples/baz\" \"TestAll/fixtures/qux\" \"TestBar\" \"TestFoo\" \"all\"]",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			if tt.intTestDir == "" {
+				tt.intTestDir = path.Join(testDirWithDiscovery, intTestDir)
+			}
+			err := isValidTestName(tt.intTestDir, tt.testName)
+			if tt.errMsg != "" {
+				assert.NotNil(err)
+				assert.Contains(err.Error(), tt.errMsg)
+			} else {
+				assert.NoError(err)
+			}
+		})
+	}
+}
+
+func TestGetTestCmd(t *testing.T) {
+
+	tests := []struct {
+		name       string
+		intTestDir string
+		testStage  string
+		testName   string
+		wantArgs   []string
+		errMsg     string
+	}{
+		{
+			name:     "single test",
+			testName: "TestFoo",
+			wantArgs: []string{"./...", "-run", "TestFoo", "-p", "1", "-count", "1", "-timeout", "0"},
+		},
+		{
+			name:     "all tests",
+			testName: "all",
+			wantArgs: []string{"./...", "-p", "1", "-count", "1", "-timeout", "0"},
+		},
+		{
+			name:      "custom stage",
+			testName:  "TestFoo",
+			testStage: "init",
+			wantArgs:  []string{"./...", "-run", "TestFoo", "-p", "1", "-count", "1", "-timeout", "0"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			if tt.intTestDir == "" {
+				tt.intTestDir = path.Join(testDirWithDiscovery, intTestDir)
```

Comment:
We're skipping the automatic discovery from working directory here. Would be good to make sure there's coverage for that.

Reply:
Since its hard to mock working directory, I have pulled the discovery into a separate function and tested here. 
https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/34ea0afc6f65161479af36a101937567c6012ef3/cli/bptest/list_test.go#L188-L212